modelon/impact/client/options.py

```diff
@@ -65,26 +65,26 @@ class SolverOptions(BaseExecutionOptions):
 
 
 class SimulationOptions(BaseExecutionOptions):
-    def with_result_filter(self, pattern: Union[str, List[str]]) -> SimulationOptions:
+    def with_result_filter(self, pattern: List[str]) -> SimulationOptions:
         """Sets the variable filter for results.
 
         Args:
             pattern:
-                A filter pattern for choosing which variables to actually
+                A list of filter patterns for choosing which variables to actually
                 store result for. The syntax can be found in
                 https://en.wikipedia.org/wiki/Glob_(programming).
                 An example for filter is `*der`, which would stores all
-                variables ending with `der`. Can also be a list.
+                variables ending with `der`.
 
         Example::
 
             sim_opts = custom_function.get_simulation_options().with_result_filter(
                 pattern = ["*.phi"])
 
         """
-        if not isinstance(pattern, str):
-            pattern = str(pattern)
-        return self.with_values(filter=pattern)
+        if not isinstance(pattern, list):
```

Comment:
Let's do it all the way: `not isinstance(pattern, list) or [p for p in pattern if not isinstance(p, str)]` and also add test

Reply:
Thanks @iakovn ! Fixed it :)